File: trabalho/Solucao/Exercicio_2/Banco/src/main/java/banco/dominio/excecoes/ExcecaoChequeEspecial.java
package banco.dominio.excecoes;

public class ExcecaoChequeEspecial extends Exception{
    private double deficit;

    public ExcecaoChequeEspecial(String mensagem, double deficit) {
        super(mensagem);
        this.deficit = deficit;
    }

    public double getDeficit() {
        return deficit;
    }
}

File: javascript/ExcecaoChequeEspecial.js
module.exports = class ExcecaoChequeEspecial {
  constructor(mensagem, deficit) {
    this.mensagem = mensagem;
    this.deficit = deficit;
  }

  getDeficit() {
    return this.deficit;
  }

  getMessage() {
    return this.mensagem;
  }
};
